File: Backend/Validation/ReviewRules.cs
namespace Backend.Validation;

public static class ReviewRules
{
    public const int CommentMaxLength = 500;
    public const string CommentRequired = "Vui lòng nhập nhận xét khi đánh giá từ 1 đến 3 sao.";
    public const string CommentTooLong = "Nhận xét không được vượt quá 500 ký tự.";

    public static string? NormalizeComment(string? comment)
    {
        var trimmed = comment?.Trim();
        return string.IsNullOrEmpty(trimmed) ? null : trimmed;
    }

    public static string? ValidateComment(byte rating, string? comment, out string? normalizedComment)
    {
        normalizedComment = NormalizeComment(comment);

        if (rating is >= 1 and <= 3 && normalizedComment is null)
            return CommentRequired;

        if (normalizedComment is { Length: > CommentMaxLength })
            return CommentTooLong;

        return null;
    }
}

File: CustomerApp/lib/utils/review_rules.dart
class ReviewRules {
  static const int commentMaxLength = 500;
  static const String commentRequired =
      'Vui lòng nhập nhận xét khi đánh giá từ 1 đến 3 sao.';
  static const String commentTooLong = 'Nhận xét không được vượt quá 500 ký tự.';

  static String? normalizeComment(String? comment) {
    final trimmed = comment?.trim();
    if (trimmed == null || trimmed.isEmpty) return null;
    return trimmed;
  }

  static String? validateComment(int rating, String? comment) {
    final normalized = normalizeComment(comment);
    if (rating >= 1 && rating <= 3 && normalized == null) {
      return commentRequired;
    }
    if (normalized != null && normalized.length > commentMaxLength) {
      return commentTooLong;
    }
    return null;
  }
}
